java/lang/Double/Double.cpp

```diff
@@ -358,209 +354,208 @@ double Double::min(double doubleA, double doubleB) {
-		index--;
-		integerPartDoubleInput = integerPartDoubleInput >> 1;
-	}
-	
-	fractionPartDoubleInput = doubleInput - integerPartDoubleInput;
-	index = 0;
-	
-	while ((fractionPartDoubleInput != 0) && (index < 24)) {
-		fractionPartDoubleInput = fractionPartDoubleInput * 2;
-		
-		int integerPart = (int) floor(fractionPartDoubleInput);
-		
-		if (integerPart == 1) {
-			fractionPartNormalizeForm[index] = '1';
-		}
-		
-		if (integerPart == 0) {
-			fractionPartNormalizeForm[index] = '0';
-		}
-		
-		index++;
-		fractionPartDoubleInput = fractionPartDoubleInput - integerPart;
-	}
-	
-	sizeOfFractionPartNormalizeForm = index;
-	
-	/** Copy value from integerPartNormalizeForm to doubleInputNormalizeForm */
-	for( i = 0; i < sizeOfIntegerPartNormalizeForm; i++ ) {
-		doubleInputNormalizeForm[i] = integerPartNormalizeForm[i];
-	}
-	
-	indexOfDotDoubleInputNormalizeForm = sizeOfIntegerPartNormalizeForm;
-	
-	doubleInputNormalizeForm[indexOfDotDoubleInputNormalizeForm] = '.';
-	
-	/** Copy value from fractionPartNormalizeForm
-	 * to doubleInputNormalizeForm
-	 */
-	index = indexOfDotDoubleInputNormalizeForm +1;
-	for( i = 0; i <= sizeOfFractionPartNormalizeForm; i++, index++ ) {
-		doubleInputNormalizeForm[index]
-			= fractionPartNormalizeForm[i];
-	}
-	
-	sizeOfDoubleInputNormalizeForm = sizeOfIntegerPartNormalizeForm + 1
-	                                 + sizeOfFractionPartNormalizeForm;
-	
-	indexFirstBit1DoubleInputNormalizeForm = 1;
-	for ( i = 0; i <= sizeOfDoubleInputNormalizeForm; i++ ) {
-		if (doubleInputNormalizeForm[i] == '1') {
-			indexFirstBit1DoubleInputNormalizeForm = i;
-			break;
-		}
-	}
-	
-	integerPartDoubleInput =  (int) floor(doubleInput);
-	if (integerPartDoubleInput != 0) {
-		powerExponentBase2 = indexOfDotDoubleInputNormalizeForm
-		                     - indexFirstBit1DoubleInputNormalizeForm -1;
-	}
-	
-	if (integerPartDoubleInput == 0) {
-		powerExponentBase2 = indexOfDotDoubleInputNormalizeForm
-		                     - indexFirstBit1DoubleInputNormalizeForm;
-	}
-	
-	exponentDoubleInput = powerExponentBase2 + exponentBiasBinary32;
-	
-	index = sizeOfExponentPartBinary32;
-	
-	while (exponentDoubleInput != 0) {
-		if ( (exponentDoubleInput & 1 ) == 1) {
-			resultDoubleToBinary32StringType[index] = '1';
-		}
-		
-		if ( (exponentDoubleInput & 1 ) == 0) {
-			resultDoubleToBinary32StringType[index] = '0';
-		}
-		
-		index--;
-		exponentDoubleInput = exponentDoubleInput >> 1;
-	}
-	
-	indexBeginFractionPartResultDoubleToBinary32StringType
-		= sizeOfExponentPartBinary32 + 1;
-	
-	index = indexBeginFractionPartResultDoubleToBinary32StringType;
-	i = indexFirstBit1DoubleInputNormalizeForm + 1;
-	
-	for (index ; index <= 31; index++) {
-		if (i == sizeOfDoubleInputNormalizeForm) {
-			break;
-		}
-		
-		if (i == indexOfDotDoubleInputNormalizeForm) {
-			i++;
-		}
-		
-		resultDoubleToBinary32StringType[index]
-			= doubleInputNormalizeForm[i];
-		i++;
-	}
+string Double::doubleToBinary32StringType(double doubleInput) {
```

Comment:
Please use String and Math class

Reply:
Sorry. I don't understand how to use String in this method.
About Math class I think don't need to fix it.

Reply:
Why not Math class? :sob: